rts/Game/UI/KeyBindings.cpp

```diff
@@ -832,6 +846,34 @@ bool CKeyBindings::ExecuteCommand(const std::string& line)
 			debugEnabled = atoi(words[1].c_str());
 		}
 	}
+	else if (command == "keyload") {
+		const std::string filename = words.size() > 1 ? words[1] : "uikeys.txt";
+
+		if (debugEnabled)
+			LOG("[CKeyBindings::%s] line=%s", __func__, line.c_str());
+
+		if (loadStack.empty() && words.size() == 1)
+			LoadDefaults();
+
+		Load(filename);
+	}
+	else if (command == "keyreload") {
+		const std::string filename = words.size() > 1 ? words[1] : "uikeys.txt";
```

Comment:
Perhaps this could be a `string_view`?

Reply:
I'm not aware of that, can you clarify?

Reply:
I found std::string_view although it's not clear to me how that's useful, can you clarify?

Reply:
`std::string` allocates a copy, which we could avoid since we don't modify it and could use existing strings. `std::string_view` is non-owning, so it just points to either the contents of words[1] or to "uikeys.txt" without having to make a copy. It's essentially the same type of microoptimisation as the `&` above so it's perfectly fine to leave as is.

Reply:
I prefer using the pass by reference approach instead, it looks unnecessarily complicated. Thanks for the tip though, I wasn't aware of string_view, seems very useful for iterating stuff.

Reply:
If you're dealing with existing strings and want to use them as-is, `const std::string &` is just as good.

The real benefit is if you want to work with slices of a string. In particular, `CSimpleParser::Tokenize` could return a vector of string_views, where instead of the memory looking like
![image](https://user-images.githubusercontent.com/2573076/199861354-ce2007d1-bfd7-467e-bfb6-58b6163b59e5.png)
it would look like:
![image](https://user-images.githubusercontent.com/2573076/199861379-572901d3-13b5-45c5-9dfd-5e91ff0f175c.png)

Additionally, if there is no pre-existing object such as with `const std::string &foo = "foo";` it still has to make a copy (to turn the `const char *` into a full `std::string`) while `std::string_view` points to the original memory.

But again, in our case it's an unimportant function and maybe the compiler can notice we're not doing anything interesting and optimize it on its own.

Reply:
I see, I'll keep this in mind, thanks for the detailed explanation.

Reply:
Hmm. I was about to create a #good-first-issue for refactoring Tokenize to return a string_view but I read this:

```
Returning a std::string_view from a function is usually a bad idea. We’ll explore why in lesson [11.7 -- std::string_view (part 2)](https://www.learncpp.com/cpp-tutorial/stdstring_view-part-2/). For now, avoid doing so.
```

Probably because it retains the reference to a string that might be changed.

Reply:
Yeah, though in this case the underlying string isn't changed, and depending on how common that case is across the codebase it might still be worth adding something like TokenizeUnsafe with string_views in parallel to the regular Tokenize with strings, and use where applicable (at a glance it's majority, but on the other hand it's all some fairly unimportant functions).